plugins/maven/maven3-server-impl/src/org/jetbrains/idea/maven/server/Maven3XServerEmbedder.java

```diff
@@ -797,6 +807,10 @@ public Collection<MavenExecutionResult> doResolveProject(@NotNull final Collecti
 
     request.setUpdateSnapshots(myAlwaysUpdateSnapshots);
 
+    if (myServerSettings.isTychoProject()) {
+      request.setDegreeOfConcurrency(2);
```

Comment:
This can be increased up to 4, but file access doesn't seem to be thread safe.

Reply:
this should not be tycho-specific